dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java

```diff
@@ -561,11 +561,16 @@ private Userinfoplus getUserInfo(String accessToken) {
         @Authorization(value = JWT_SECURITY_DEFINITION_NAME)}, notes = "A post method is required by satellizer to send the GitHub token", response = Token.class)
     public Token addToken(@ApiParam("code") String satellizerJson) {
         Gson gson = new Gson();
-        JsonElement element = gson.fromJson(satellizerJson, JsonElement.class);
-        JsonObject satellizerObject = element.getAsJsonObject();
-        final String code = getCodeFromSatellizerObject(satellizerObject);
-        final boolean registerUser = getRegisterFromSatellizerObject(satellizerObject);
-        return handleGitHubUser(null, code, registerUser);
+        try {
+            JsonElement element = gson.fromJson(satellizerJson, JsonElement.class);
+            JsonObject satellizerObject = element.getAsJsonObject();
+            final String code = getCodeFromSatellizerObject(satellizerObject);
+            final boolean registerUser = getRegisterFromSatellizerObject(satellizerObject);
+            return handleGitHubUser(null, code, registerUser);
+        } catch (NullPointerException ex) {
```

Comment:
This is an effective way to eliminate the 500s, but if possible, it might be better to address the root cause of the `NullPointerException`.  Except in some extreme [parsing?] situations where it's excessively verbose to test for all the nulls, it's probably better if our code doesn't throw `NullPointerException`s at all.  Also, the big `try-catch` block reduces our ability to detect (via our 500 alarm) any other `NullPointerException`s which might someday spring forth from that block.  And, sometimes, the `NullPointerException` is a symptom of a bigger problem which is exposed during the analysis/repair...

Reply:
Thanks for this comment! I changed my code to add an if case to address the root cause of the `NullPointerException`s and removing the try-catch blocks.